src/cpp/src/cache_manager.hpp

```diff
@@ -89,11 +89,22 @@ class CacheManager {
                     ov::Tensor key_src_cache_roi(m_key_cache[decoder_layer_id], key_src_start_roi, key_src_end_roi);
                     ov::Tensor key_dst_cache_roi(m_key_cache[decoder_layer_id], key_dst_start_roi, key_dst_end_roi);
 
-                    ov::Tensor value_src_cache_roi(m_value_cache[decoder_layer_id], value_src_start_roi, value_src_end_roi);
-                    ov::Tensor value_dst_cache_roi(m_value_cache[decoder_layer_id], value_dst_start_roi, value_dst_end_roi);
-
                     key_src_cache_roi.copy_to(key_dst_cache_roi);
-                    value_src_cache_roi.copy_to(value_dst_cache_roi);
+                    const auto& value_cache_prec = m_value_cache[decoder_layer_id].get_element_type();
+                    if (value_cache_prec != ov::element::u4 && value_cache_prec != ov::element::i4)  {
```

Comment:
Could you please extract tensor copy for low bits data types and submit in a separate PR?